src/coresyftools/coresyftools/gpt_tool.py

```diff
@@ -0,0 +1,101 @@
+    pass
+
+class GPTExecutionException(Exception):
+    '''Error occurred during a SNAP gpt execution'''
+
+    ERROR_REGEX = re.compile('Error:')
+
+    def __init__(self, returncode, msg):
+        self.returncode = returncode
+        self.errors = [line[7:] for line in msg.split(
+            os.linesep) if self.ERROR_REGEX.match(line)]
+        super(GPTExecutionException, self).__init__(self.errors)
+
+
+class GPTGraphFileNotFound(Exception):
+    pass
+
+
+class GPTCoReSyFTool(CoReSyFTool):
+    '''CoReSyF Tool consisting of a single SNAP gpt operation.'''
+
+    DEFAULT_FORMAT = 'GeoTIFF-BigTIFF'
+    DEFAULT_EXT = 'tif'
+    DEFAULT_GPT_GRAPH_FILE_NAME = 'gpt_graph.xml'
+
+    def _validate_operation(self, operation):
+        if 'operation' not in operation and ('graph' not in operation or not operation['graph']):
+            raise InvalidManifestException('A operation or graph flag should be present.')
+        if 'operation' in operation and 'graph' in operation and operation['graph']:
+            raise InvalidManifestException('Can not be operation and graph at same time.')
+        if 'graph' in self.operation and self.operation['graph']:
+            graph_file = os.path.join(self.context_directory, self.DEFAULT_GPT_GRAPH_FILE_NAME)
+            if not os.path.exists(graph_file):
+                raise GPTGraphFileNotFound(graph_file)
+
+    def run(self, bindings):
+        if len(self.arg_parser.inputs) > 1:
+            raise TooManyInputArgumentsException()
+        if len(self.arg_parser.outputs) > 1:
+            raise TooManyInputArgumentsException()
+
+        bindings = bindings.copy()
+        operator = None
+        if 'operation' in self.operation:
+            operator = self.operation['operation']
+        elif 'graph' in self.operation and self.operation['graph']:
+            graph_file = os.path.join(self.context_directory, self.DEFAULT_GPT_GRAPH_FILE_NAME)
+            if not os.path.exists(graph_file):
+                raise GPTGraphFileNotFound(graph_file)
+            operator = graph_file
+        if 'parameters' in self.operation:
+            parameters = self.operation['parameters']
+            bindings.update(parameters)
+        source = bindings.pop(self.arg_parser.inputs[0])
+        target = bindings.pop(self.arg_parser.outputs[0])
+        self._call_gpt(operator, source, target, bindings)
+        # This is needed because SNAP automatically adds file extensions, but
+        # output files can not have a name different from the specified in
+        # the command line.
+        self._remove_snap_file_extension(target)
+
+    def _call_gpt(self, operator, source, target, options):
+        source = os.path.abspath(source)
+        target = os.path.abspath(target)
+        args = ['gpt', operator, '-f', self.DEFAULT_FORMAT,
+                '-t', target]
+        args.extend([self._option_str(arg, value)
+                     for arg, value in options.items()])
+        args.append(source)
```

Comment:
We could create a function `build_gpt_shell_command` where we would define these arguments.

Reply:
Fixed.